api: make /v1/subscribe update customer info

bmizerany opened the pull request:
This commit also removes the constraint that /v1/subscribe must
contain at least one phase.

isaacs commented:
Found a panic:

```
$ tier subscribe --email org:test
panic: runtime error: slice bounds out of range [1:0]

goroutine 1 [running]:
main.runTier({0x20576021d, 0x9}, {0xc0000160a0?, 0x2, 0x2})
	/Users/isaacs/dev/tierrun/tier/cmd/tier/tier.go:261 +0x2e3a
main.main()
	/Users/isaacs/dev/tierrun/tier/cmd/tier/tier.go:69 +0x165
```

bmizerany commented:
> Found a panic:
> 
> ```
> $ tier subscribe --email org:test
> panic: runtime error: slice bounds out of range [1:0]
> 
> goroutine 1 [running]:
> main.runTier({0x20576021d, 0x9}, {0xc0000160a0?, 0x2, 0x2})
> 	/Users/isaacs/dev/tierrun/tier/cmd/tier/tier.go:261 +0x2e3a
> main.main()
> 	/Users/isaacs/dev/tierrun/tier/cmd/tier/tier.go:69 +0x165
> ```

Resolved. Please try again.

bmizerany committed:
api,cmd/tier: accept extra org info on schedule

This commit makes subscribe accept additional informtion about an org to
attach to it when creating or updating schedules in stripe. The new
information is:

- description
- name
- phone
- email
- metadata

Metadata containing any key prefixed with ("tier.") is an error and the
whole request is rejected.

This commit also makes /v1/whois respond with all up-to-date information
attached to an orgs underlying Stripe customer if the include=info query
param is present. The response including info does not expose any
metadata added by Tier.
